pycheribuild/projects/cross/juliet_test_suite.py

```diff
@@ -94,6 +103,12 @@ def run_tests(self):
             args.append("--ld-preload-path")
             args.append(self.ld_preload_path)
 
+        for cmd in self.cwe_setup_commands:
+            args.append("--test-setup-command=" + cmd)
+
+        # For stdin redirection
+        args.append("--test-setup-command=" + "touch /tmp/in.txt")
```

Comment:
You can't just write to fixed paths in /tmp. It's unsafe. You need to create a proper temporary file or directory and add hooks to clean it up on exit. You should be able to do this with tempfile.mkstemp().

Reply:
Ah, in that case, this is fine. I missed that it was inside a on-off qemu instance.